src/freenas/usr/local/bin/truenas-autotune.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+import sys
+from types import SimpleNamespace
+
+import psutil
+
+from middlewared.plugins.system.dmi import DMIDecode
+from middlewared.plugins.system.product import is_enterprise_ix_hardware
+from middlewared.plugins.truenas import get_chassis_hardware
+from middlewared.plugins.tunables import zfs_parameter_value
+from middlewared.utils.db import query_table, update_table
+
+KiB = 1024 ** 1
+MiB = 1024 ** 2
+GiB = 1024 ** 3
+
+MIN_ZFS_RESERVED_MEM = 1 * GiB
+
+zfs_parameters = {}
+
+
+def zfs_parameter(tunable_name):
+    def decorator(func):
+        zfs_parameters[tunable_name] = func
+        return func
+
+    return decorator
+
+
+@zfs_parameter("zfs_dirty_data_max_max")
+def guess_vfs_zfs_dirty_data_max_max(context):
+    if context.hardware.startswith("M"):
```

Comment:
Same thing here. This needs to only be applied to our M50 and M60

```
context.hardware.startswith(('TRUENAS-M50', TRUENAS-M60'))

Reply:
Forgot to pre-process `hardware` variable like the previous autotune did